blockchain/validation.go

```diff
@@ -69,8 +121,51 @@ func (bc *BlockChain) ValidTransaction(t *Transaction) (bool, TransactionCode) {
 	return true, ValidTransaction
 }
 
+// ValidCloudBase returns true if a transaction is a valid CloudBase transaction
+// and false otherwise
+func ValidCloudBase(t *Transaction) (bool, CloudBaseTransactionCode) {
```

Comment:
This function should be aware of the block as well. It needs to verify that the output matches the block reward for the given block and it needs to verify that the output address is the same as the miner who mined the block.